### DISCUSSION TITLE
Main header customization (Extend nova layouts)

### DISCUSSION THREAD

**Discussion by AbdallahAli912** [laravel/nova-issues#6141] [Q&A]
- Laravel Version: 9
- Nova Version: 9

### we find a way to extends nova components but we have problem in extending layouts.
for example:  we want to customize main header  in js/layouts/AppLayout.vue but we failed in extending layouts.
<img width="1431" alt="Screenshot 2024-01-09 at 12 09 58 PM" src="https://github.com/laravel/nova-issues/assets/54299150/5e5bbeb4-330c-4419-9192-ffb28fe64263">

### Detailed steps to extend layout
1- create custom tool
2- update webpack.mix.js to get mixins,js,node modules from nova folder in vendor
3- copy layout vue file.
4- register new layout in tool.js

**Comment by pierre-elie-fortin** [comment]
You can use this maybe ? https://novapackages.com/packages/norman-huth/nova-assets-changer

**Comment by Aakashjoshi98** [comment]
I have tried to implement it ,but i'm stuck in it and not able to see any thing like you have shared 
I need one new custom drop down beside notification if anyone have any idea please let me know.

**Reply by vishwasShifl** [reply]
I'm also stuck to update header component.
